code/main.cpp

```diff
@@ -1,9 +1,20 @@
 #include "hwlib.hpp"
+#include <gas_sensors_c.hpp>
+
+template<typename T, std::size_t N>
+hwlib::ostream & operator<< (hwlib::ostream & out, const std::array<T, N> & rhs)
```

Comment:
space after & is not according to style guide, should be `&operator<<`, '&out', '&rhs'